### PR TITLE
Bump numerous dependencies

### PR DESCRIPTION
**Pull request by polomani**
This PR is to fix CVEs primarily, but also update the codebase where required.

- migrate to @grafana/create-plugin from @grafana/toolkit (new recommended way for plugin dev)
- all files in `.config/*` are generated by @grafana/create-plugin

Those files are also generated by @grafana/create-plugin:
- .eslintrc
- .nvmrc
- .prettierrc.js
- docker-compose.yaml
- jest-setup.js
- jest.config.js
- tsconfig.json

Dependencies and commands in package.json are updated by the script as well.

**Actual changes:**
- added `getDataSourceWithMocks` so we can initiate data source for every describe(..), the previous approach doesn't seem to work with the new setup
- moved test_utils outside of __tests__ folder
-  updated "@grafana/data": "^9.4.7", "@grafana/runtime": "^9.4.7", "@grafana/ui": "^9.4.7"

This needs to be tested with Grafana v8 / v9

Bumped Node to v16 in GitHub actions.

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+version: '3.0'
+
+services:
+  grafana:
+    container_name: 'cognitedata-datasource'
+    build:
+      context: ./.config
+      args:
+        grafana_version: ${GRAFANA_VERSION:-9.3.8}
```
Comment (polomani) [inline docker-compose.yaml:9]: we should try this with v8.5~ and 9.X

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -20,6 +20,9 @@ jobs:
       - name: Install dependencies
         run: yarn install
 
+      - name: Test plugin
+        run: yarn test:ci
+
```
Comment (polomani) [inline .github/workflows/release.yml:25]: we need to sign the plugin as well (change the command)

Comment (polomani) [reply .github/workflows/release.yml:25]: update the whole file

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -20,6 +20,9 @@ jobs:
       - name: Install dependencies
```
Comment (polomani) [inline .github/workflows/release.yml:20]: use node 16

**Review by abdullah-cognite — APPROVED** [review]

**Pull request merged**